controller/api/models.py

```diff
@@ -343,7 +343,7 @@ def deploy(self, user, release, initial=False):
             self._destroy_containers(existing)
 
         # perform default scaling if necessary
-        if initial:
+        if self.structure == {} and release.build is not None:
```

Comment:
This is intentional so that users who do a `deis config:set` before they do an intial deploy won't blow up on them. If you remove that call, you'll end up with the errors listed above. This is because `_default_scale` requires that the release has a non-null build object.